packages/renderer/MdToHtml/linkReplacement.ts

```diff
@@ -27,16 +28,14 @@ export interface LinkReplacementResult {
 }
 
 export default function(href: string, options: Options = null): LinkReplacementResult {
-	options = {
-		title: '',
-		resources: {},
-		ResourceModel: null,
-		linkRenderingType: 1,
-		plainResourceRendering: false,
-		postMessageSyntax: 'postMessage',
-		enableLongPress: false,
-		...options,
-	};
+	options = { ...options };
+	options.title ??= '';
+	options.resources ??= {};
+	options.ResourceModel ??= null;
+	options.linkRenderingType ??= LinkRenderingType.JavaScriptHandler;
+	options.plainResourceRendering ??= false;
+	options.postMessageSyntax ??= 'postMessage';
+	options.enableLongPress ??= false;
```

Comment:
Previously, `options.linkRenderingType` was `undefined` in some cases. With the previous method of setting default property values, `options.linkRenderingType` would remain `undefined`, rather than using the default value.

Reply:
Is it not possible to ensure that we don't set `linkRenderingType` to `undefined` anywhere?

Reply:
Currently, `linkRenderingType` is set to `undefined` here (and possibly elsewhere):

https://github.com/laurent22/joplin/blob/147a66d64e74ae1c5643fffc647f0be4bec854ef/packages/renderer/MdToHtml/rules/link_open.ts#L17-L21

While there are ways to rewrite `link_open.ts`'s usage of `linkReplacement` to not set `linkRenderingType` to `undefined`, these may be more complicated than the above change. For example,
- Writing a function to set properties on the options object only if not `undefined`. For example,
   ```typescript
 	const replacement = linkReplacement(href, objectAssignNonUndefined({}, { 
 	    title, 
 	    resources: ruleOptions.resources, 
 	    ResourceModel: ruleOptions.ResourceModel, 
 	    linkRenderingType: ruleOptions.linkRenderingType,
 	}));
   ```
   where
   ```typescript
   // Pseudocode:
   const objectAssignNonUndefined = <T extends object> (target: T, source: T): T => {
      result = {...target};
      for (const prop in source) {
         if (source[prop] !== undefined) {
            result[prop] = source[prop];
         }
      }
      return result;
   };
   ```
- Setting defaults when calling `linkReplacement`. For example,
  ```typescript
 	const replacement = linkReplacement(href, { 
 	    title: title ?? '', 
 	    resources: ruleOptions.resources ?? {}, 
 	    ResourceModel: ruleOptions.ResourceModel ?? null, 
 	    linkRenderingType: ruleOptions.linkRenderingType ?? LinkRenderingType.JavaScript,
 	});
   ```

**Other ways to handle the case where `linkRenderingType` is `undefined`**: Here are a two alternatives to the syntax used in this pull request:
1. Destructuring. For example,
  ```typescript
	export default function(href: string, {
		title = '',
		resources = {},
		ResourceModel = null,
		linkRenderingType = LinkRenderingType.JavaScriptHandler,
		plainResourceRendering = false,
		postMessageSyntax = 'postMessage',
		enableLongPress = false,
  	}: Options = {}): LinkReplacementResult {
  	...
  ```
  See [this section](https://google.github.io/styleguide/tsguide.html#features-objects-destructuring) of the Google TypeScript style guide.
2. Extracting and setting the default value for just `linkRenderingType`. For example,
   ```typescript
 	options = {
		title: '',
		resources: {},
		ResourceModel: null,
		plainResourceRendering: false,
		postMessageSyntax: 'postMessage',
		enableLongPress: false,
		...options,
	};
	const linkRenderingType = options.linkRenderingType ?? LinkRenderingType.JavaScriptHandler;
   ```

Reply:
Thanks for the information. It sounds like your implementation is what makes the most sense and is simpler, so let's go with this.